### ISSUE TITLE
Client stopped syncing (block-tx-server.c(631): Recv block timeout)

### ISSUE THREAD

**Issue by Pro** [haiwen/seafile#1013]
## Versions
Server: Seafile 4.0.1 on Ubuntu 14.04.1
Client: Seafile 4.0.5 on Windows 8.1 x64

## Problem description
I've a large and a smaller library which are synced between two PC (Notebook & Desktop). I'm the only user of these libraries.
Since a few days my Notebook doesn't sync the libraries anymore. I think it is related to the error message `block-tx-server.c(631): Recv block timeout`. (see logs below)
After I start the client, it tries to upload a few data packages but then stops and says all the libraries are up to date (and clearly they aren't). When I restart the client, the same happens again.

@killing there was a similar issue: #249 but there's no special solution in this issue.
How can I fix this problem?

### Server logs

I cleaned all the logfiles and restarted the server and client.

#### seafile.log (Server)
```
[01/07/2015 09:32:02 PM] ../common/mq-mgr.c(60): [mq client] mq cilent is started
[01/07/2015 09:32:02 PM] ../common/mq-mgr.c(106): [mq mgr] publish to hearbeat mq: seaf_server.heartbeat
[01/07/2015 09:32:02 PM] listen-mgr.c(115): listen on port 12001 for block tranfer
[01/07/2015 09:32:58 PM] block-tx-server.c(631): Recv block timeout.
[01/07/2015 09:34:17 PM] block-tx-server.c(631): Recv block timeout.
```

#### ccnet.log (Server)
```
[01/07/15 21:32:01] ../common/session.c(432): socket file exists, delete it anyway
[01/07/15 21:32:01] ../common/session.c(461): Listen on /var/www/seafile/ccnet/ccnet.sock for local clients
[01/07/15 21:32:01] ../common/session.c(267): Update pubinfo file
[01/07/15 21:32:01] ../common/connect-mgr.c(515): Opened port 10001 to listen for incoming peer connections
[01/07/15 21:32:02] ../common/session.c(375): Accepted a local client
[01/07/15 21:32:02] ../common/session.c(375): Accepted a local client
[01/07/15 21:32:02] ../common/session.c(375): Accepted a local client
[01/07/15 21:32:02] ../common/session.c(375): Accepted a local client
[01/07/15 21:32:04] ../common/connect-mgr.c(266): Unknown peer c5c02ef46fc4be7bbdb8638a9cffe67bd3c29337 connecting
[01/07/15 21:32:05] ../common/session.c(375): Accepted a local client
[01/07/15 21:32:05] ../common/peer.c(943): Local peer down
[01/07/15 21:32:18] ../common/session.c(375): Accepted a local client
[01/07/15 21:32:18] ../common/session.c(375): Accepted a local client
[01/07/15 21:32:19] ../common/session.c(375): Accepted a local client
[01/07/15 21:32:19] ../common/session.c(375): Accepted a local client
[01/07/15 21:32:20] ../common/connect-mgr.c(266): Unknown peer 17fd0f4b040aa8c0b2d097b3e8df8de1c1d15dd0 connecting
[01/07/15 21:32:22] ../common/session.c(375): Accepted a local client
[01/07/15 21:32:59] ../common/session.c(375): Accepted a local client
[01/07/15 21:34:26] ../common/peer.c(941): libevent got an error! what=33, errno=104 (Connection reset by peer)
[01/07/15 21:34:26] ../common/peer.c(949): [Net Error] Peer Stefan@ (17fd0f4b04) down
[01/07/15 21:34:26] ../common/processor.c(218): [Proc] Shutdown processor keepalive2-proc(-1001) for bad update: 515 peer down
[01/07/15 21:34:26] ../common/processors/keepalive2-proc.c(475): [Keepalive] Error from peer 515 peer down
```

#### controller.log (Server)
```
[01/07/15 21:32:01] seafile-controller.c(154): starting ccnet-server ...
[01/07/15 21:32:01] seafile-controller.c(73): spawn_process: ccnet-server -c /var/www/seafile/ccnet -f /var/www/seafile/logs/ccnet.log -d -P /var/www/seafile/pids/ccnet.pid
[01/07/15 21:32:01] seafile-controller.c(88): spawned ccnet-server, pid 19539
[01/07/15 21:32:02] seafile-controller.c(556): ccnet daemon connected.
[01/07/15 21:32:02] seafile-controller.c(185): starting seaf-server ...
[01/07/15 21:32:02] seafile-controller.c(73): spawn_process: seaf-server -c /var/www/seafile/ccnet -d /home/seafile -l /var/www/seafile/logs/seafile.log -P /var/www/seafile/pids/seaf-server.pid
[01/07/15 21:32:02] seafile-controller.c(88): spawned seaf-server, pid 19545
[01/07/15 21:32:02] seafile-controller.c(529): seafdav not enabled.
```

#### seafile.log (Client)
```
[01/07/15 21:32:18] seaf-daemon.c(490): starting seafile client 4.0.5
[01/07/15 21:32:18] seaf-daemon.c(492): seafile source code version 57f2ba8fc2a4e1ce0f99713cde3670254baf1268
[01/07/15 21:32:18] ../common/mq-mgr.c(60): [mq client] mq cilent is started
[01/07/15 21:32:18] ../common/mq-mgr.c(106): [mq mgr] publish to hearbeat mq: seafile.heartbeat
[01/07/15 21:32:20] sync-mgr.c(569): Repo 'Picasa' sync state transition from 'synchronized' to 'uploading'.
[01/07/15 21:32:20] sync-mgr.c(569): Repo 'Daten' sync state transition from 'synchronized' to 'downloading'.
[01/07/15 21:32:20] transfer-mgr.c(310): Transfer repo '09f896a7': ('normal', 'init') --> ('normal', 'check')
[01/07/15 21:32:20] transfer-mgr.c(310): Transfer repo 'a6fc8b71': ('normal', 'init') --> ('normal', 'check')
[01/07/15 21:32:20] processors/check-tx-v3-proc.c(313): repo version is 1, protocol version is 7.
[01/07/15 21:32:20] transfer-mgr.c(310): Transfer repo '09f896a7': ('normal', 'check') --> ('normal', 'commit')
[01/07/15 21:32:20] transfer-mgr.c(310): Transfer repo '09f896a7': ('normal', 'commit') --> ('normal', 'fs')
[01/07/15 21:32:20] processors/check-tx-v3-proc.c(313): repo version is 1, protocol version is 7.
[01/07/15 21:32:20] transfer-mgr.c(310): Transfer repo 'a6fc8b71': ('normal', 'check') --> ('normal', 'commit')
[01/07/15 21:32:20] transfer-mgr.c(310): Transfer repo 'a6fc8b71': ('normal', 'commit') --> ('normal', 'fs')
[01/07/15 21:32:20] transfer-mgr.c(310): Transfer repo 'a6fc8b71': ('normal', 'fs') --> ('normal', 'check-blocks')
[01/07/15 21:32:21] transfer-mgr.c(310): Transfer repo 'a6fc8b71': ('normal', 'check-blocks') --> ('normal', 'get-chunk-server')
[01/07/15 21:32:21] transfer-mgr.c(310): Transfer repo 'a6fc8b71': ('normal', 'get-chunk-server') --> ('normal', 'data')
[01/07/15 21:32:26] transfer-mgr.c(310): Transfer repo '09f896a7': ('normal', 'fs') --> ('normal', 'get-chunk-server')
[01/07/15 21:32:26] transfer-mgr.c(310): Transfer repo '09f896a7': ('normal', 'get-chunk-server') --> ('normal', 'data')
[01/07/15 21:32:57] block-tx-client.c(759): Data connection is closed by the server.
[01/07/15 21:33:25] transfer-mgr.c(310): Transfer repo 'a6fc8b71': ('normal', 'data') --> ('normal', 'update-branch')
[01/07/15 21:33:25] transfer-mgr.c(310): Transfer repo 'a6fc8b71': ('normal', 'update-branch') --> ('finished', 'finished')
[01/07/15 21:33:25] sync-mgr.c(569): Repo 'Picasa' sync state transition from 'uploading' to 'initializing'.
[01/07/15 21:33:26] sync-mgr.c(1385): Removing blocks for repo Picasa(a6fc8b71).
[01/07/15 21:33:26] sync-mgr.c(569): Repo 'Picasa' sync state transition from 'synchronized' to 'committing'.
[01/07/15 21:33:29] repo-mgr.c(1225): Creating partial commit after adding .
[01/07/15 21:33:29] repo-mgr.c(1241): Remain files for .
[01/07/15 21:33:29] sync-mgr.c(569): Repo 'Picasa' sync state transition from 'committing' to 'uploading'.
[01/07/15 21:33:30] transfer-mgr.c(310): Transfer repo 'a6fc8b71': ('normal', 'init') --> ('normal', 'check')
[01/07/15 21:33:30] processors/check-tx-v3-proc.c(313): repo version is 1, protocol version is 7.
[01/07/15 21:33:30] transfer-mgr.c(310): Transfer repo 'a6fc8b71': ('normal', 'check') --> ('normal', 'commit')
[01/07/15 21:33:30] transfer-mgr.c(310): Transfer repo 'a6fc8b71': ('normal', 'commit') --> ('normal', 'fs')
[01/07/15 21:33:30] transfer-mgr.c(310): Transfer repo 'a6fc8b71': ('normal', 'fs') --> ('normal', 'check-blocks')
[01/07/15 21:33:31] transfer-mgr.c(310): Transfer repo 'a6fc8b71': ('normal', 'check-blocks') --> ('normal', 'get-chunk-server')
[01/07/15 21:33:31] transfer-mgr.c(310): Transfer repo 'a6fc8b71': ('normal', 'get-chunk-server') --> ('normal', 'data')
[01/07/15 21:33:42] block-tx-client.c(935): Recv timeout.
```

#### ccnet.log (Client)
```
[01/07/15 21:32:17] ccnet-daemon.c(193): starting ccnet client 4.0.5
[01/07/15 21:32:17] ccnet-daemon.c(195): ccnet source code version 6b8e8718825af85f481b7da61e919d848af2d56d
[01/07/15 21:32:17] ../common/session.c(395): Listen on 127.0.0.1 13419
[01/07/15 21:32:17] ../common/session.c(267): Update pubinfo file
[01/07/15 21:32:18] ../common/session.c(375): Accepted a local client
[01/07/15 21:32:18] ../common/session.c(375): Accepted a local client
[01/07/15 21:32:18] ../common/session.c(375): Accepted a local client
[01/07/15 21:32:18] ../common/session.c(375): Accepted a local client
[01/07/15 21:32:18] ../common/session.c(375): Accepted a local client
[01/07/15 21:32:18] ../common/session.c(375): Accepted a local client
[01/07/15 21:32:18] ../common/processors/rcvcmd-proc.c(492): Add server 4e9d5169 seafile.example.com:10001
[01/07/15 21:32:18] ../common/connect-mgr.c(364): [Conn] Start outgoing connect to (null)(4e9d5169b9) [** hidden IP **]:10001
[01/07/15 21:32:18] ../common/connect-mgr.c(289): [Conn] Peer (null) (4e9d5169b9) connected
[01/07/15 21:32:21] ../common/peer.c(766): Delayed update from local-2(17fd0f4b), id is 1012, 500 Shutdown
[01/07/15 21:32:26] ../common/peer.c(766): Delayed update from local-2(17fd0f4b), id is 1013, 500 Shutdown
[01/07/15 21:33:31] ../common/peer.c(766): Delayed update from local-2(17fd0f4b), id is 1020, 500 Shutdown
[01/07/15 21:33:45] ../common/peer.c(941): libevent got an error! what=33, errno=0 (No error)
[01/07/15 21:33:45] ../common/peer.c(943): Local peer down
[01/07/15 21:33:45] ../common/peer.c(941): libevent got an error! what=33, errno=0 (No error)
[01/07/15 21:33:45] ../common/peer.c(943): Local peer down
[01/07/15 21:33:45] ../common/peer.c(941): libevent got an error! what=33, errno=0 (No error)
[01/07/15 21:33:45] ../common/peer.c(943): Local peer down
[01/07/15 21:33:45] ../common/processor.c(218): [Proc] Shutdown processor rpcserver-proc(-1001) for bad update: 515 peer down
[01/07/15 21:33:45] ../common/processor.c(218): [Proc] Shutdown processor mqserver-proc(-1002) for bad update: 515 peer down
[01/07/15 21:33:45] ../common/processor.c(292): [Proc] peer 17fd0f4b04, Shutdown processor service-stub-proc(1001) for bad response: 515 peer down
[01/07/15 21:33:45] ../common/processor.c(292): [Proc] peer 17fd0f4b04, Shutdown processor service-proxy-proc(-1001) for bad response: 515 peer down
[01/07/15 21:33:46] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:46] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:46] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:46] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:46] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:47] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:47] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:47] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:47] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:47] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:48] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:48] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:48] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:48] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:48] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:49] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:49] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:49] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:49] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:49] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:50] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:50] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:50] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:50] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:50] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:51] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:51] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:51] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:51] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:51] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:52] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:52] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:52] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:52] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:52] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:53] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
[01/07/15 21:33:53] ../common/peer.c(766): Delayed update from local-4(17fd0f4b), id is 1001, 301 CLIENT CALL
... and so on ...
```

#### applet.log (Client)
```
[01/07/15 21:33:46][Sea RPC] Bad response: 515 peer down.
[01/07/15 21:33:46][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:46][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:46][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:46][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:47][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:47][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:47][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:47][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:47][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:48][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:48][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:48][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:48][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:48][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:49][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:49][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:49][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:49][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:49][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:50][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:50][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:50][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:50][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:50][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:51][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:51][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:51][Sea RPC] Bad response: 102 processor is dead.
[01/07/15 21:33:51][Sea RPC] Bad response: 102 processor is dead.
... and so on ...
```

**Comment by shoeper** [comment]
First I'd try to update server to 4.0.4

**Comment by freeplant** [comment]
Please try v4.0.6

**Comment by Pro** [comment]
Seems to be fixed with Server 4.0.4 and Client 4.0.6
THANKS!
